pkg/sbom/core/bom.go

```diff
@@ -238,12 +238,20 @@ func (b *BOM) AddComponent(c *Component) {
 }
 
 func (b *BOM) AddRelationship(parent, child *Component, relationshipType RelationshipType) {
+	// Check the wrong parent to avoid `panic`
+	if parent == nil {
+		return
+	}
```

Comment:
I added parent check to SPDX (like we do for CycloneDX). But I added 1 more check here so that we don't panic if we use this function later somewhere else.

But it might make sense to add log message here.
@knqyf263 wdyt?

Reply:
Looks good